kmath-core/src/commonMain/kotlin/scientifik/kmath/operations/Complex.kt

```diff
@@ -8,28 +8,47 @@ import scientifik.memory.MemorySpec
 import scientifik.memory.MemoryWriter
 import kotlin.math.*
 
+/**
+ * A complex conjugate.
+ */
+val Complex.conjugate: Complex
+    get() = Complex(re, -im)
+
+/**
+ * Absolute value of complex number.
+ */
+val Complex.r: Double
+    get() = sqrt(re * re + im * im)
+
+/**
+ * An angle between vector represented by complex number and X axis.
+ */
+val Complex.theta: Double
+    get() = atan(im / re)
+
 private val PI_DIV_2 = Complex(PI / 2, 0)
 
 /**
- * A field for complex numbers
+ * A field for complex numbers.
  */
 object ComplexField : ExtendedField<Complex> {
-    override val zero: Complex = Complex(0.0, 0.0)
+    override val zero: Complex = Complex(0, 0)
+    override val one: Complex = Complex(1, 0)
 
-    override val one: Complex = Complex(1.0, 0.0)
-
-    val i = Complex(0.0, 1.0)
+    /**
+     * The imaginary unit constant.
+     */
+    val i = Complex(0, 1)
 
     override fun add(a: Complex, b: Complex): Complex = Complex(a.re + b.re, a.im + b.im)
-
     override fun multiply(a: Complex, k: Number): Complex = Complex(a.re * k.toDouble(), a.im * k.toDouble())
 
     override fun multiply(a: Complex, b: Complex): Complex =
         Complex(a.re * b.re - a.im * b.im, a.re * b.im + a.im * b.re)
 
     override fun divide(a: Complex, b: Complex): Complex {
-        val norm = b.re * b.re + b.im * b.im
-        return Complex((a.re * b.re + a.im * b.im) / norm, (a.re * b.im - a.im * b.re) / norm)
+        val scale = b.re * b.re + b.im * b.im
+        return a * Complex(b.re / scale, -b.im / scale)
```

Comment:
You are creating additional intermediate Complex value here.

Reply:
However, the previous implementation was just invalid

Reply:
Maybe, you need to add a test for it.

Reply:
Tests are added